src/antigravity/memory.py

```
@@ -0,0 +1,259 @@
+class MemoryManager:
+    """
+    Manages agent memory with recursive summarization.
+
+    Features:
+    - Token counting and limit management
+    - Automatic summarization when threshold reached
+    - Persistent storage to file
+    - Context reconstruction for agent
+    """
+
+    def __init__(
```

Ruff (0.14.8)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)